src/main/java/com/segment/analytics/android/integrations/optimizelyx/OptimizelyXIntegration.java

```diff
@@ -1,63 +1,107 @@
+ * @see <a href="https://www.optimizely.com/">Optimizely</a>
+ * @see <a href="https://segment.com/docs/destinations/optimizelyx/">Optimizely Integration</a>
+ * @see <a
+ *     href="https://developers.optimizely.com/x/solutions/sdks/introduction/index.html?language=android&platform=mobile">Optimizely
+ *     Android SDK</a>
  */
 public class OptimizelyXIntegration extends Integration<Void> {
-  private static final String OPTIMIZELY_KEY = "Optimizely";
 
-  private final Context context;
+  private static final String OPTIMIZELYX_KEY = "Optimizely X";
+  final NotificationListener listener;
+  private final OptimizelyClient client;
   private final Logger logger;
-  private final Analytics analytics;
-
-  // Optimizely must be initialized immediately on launch. So we require the token when creating the
-  // factory, and initialize it in this method.
-//  public static Factory createFactory(final Application application, String token) {
-////    Optimizely.startOptimizelyWithAPIToken(token, application);
-////    return new Factory() {
-////      @Override public Integration<?> create(ValueMap settings, Analytics analytics) {
-////        boolean listen = settings.getBoolean("listen", false);
-////        Logger logger = analytics.logger(OPTIMIZELY_KEY);
-////
-////        OptimizelyIntegration integration =
-////            new OptimizelyIntegration(analytics, application, logger);
-////        if (listen) {
-////          Optimizely.addOptimizelyEventListener(integration);
-////        }
-////        return integration;
-////      }
-//
-////      @Override public String key() {
-////      return OPTIMIZELY_KEY;
-//////      }
-//    };
-//  }
-
-  public OptimizelyXIntegration(Analytics analytics, Context context, Logger logger) {
-    this.analytics = analytics;
-    this.context = context;
+  static final Options options = new Options().setIntegration(OPTIMIZELYX_KEY, false);
+
+  public static Factory factory(OptimizelyClient client) {
+    return new Factory(client);
+  }
+
+  private static class Factory implements Integration.Factory {
+    private final OptimizelyClient client;
+
+    Factory(OptimizelyClient client) {
+      this.client = client;
+    }
+
+    @Override
+    public Integration<?> create(ValueMap settings, Analytics analytics) {
+      Logger logger = analytics.logger(OPTIMIZELYX_KEY);
+      return new OptimizelyXIntegration(analytics, client, logger);
+    }
+
+    @Override
+    public String key() {
+      return OPTIMIZELYX_KEY;
+    }
+  }
+
+  public OptimizelyXIntegration(final Analytics analytics, OptimizelyClient client, Logger logger) {
+    this.client = client;
     this.logger = logger;
+
+    listener = new OptimizelyNotificationListener(analytics);
+    client.addNotificationListener(listener);
   }
 
-  @Override public void identify(IdentifyPayload identify) {
+  @Override
+  public void track(TrackPayload track) {
+    super.track(track);
+
+    if (!isNullOrEmpty(track.userId())) {
```

Comment:
We may need to consider only tracking anonymous users, since an experiment is likely to be set early on, before an `identify` is called.